default/python/AI/EspionageAI.py

```diff
@@ -112,6 +116,15 @@ def colony_detectable_by_empire(planet_id, species_name=None, empire=ALL_EMPIRES
 
     total_stealth = planet_stealth + sum([AIDependencies.STEALTH_STRENGTHS_BY_SPECIES_TAG.get(tag, 0)
                                           for tag in species_tags])
+
+    if verify_prediction:
+        meter_stealth = planet.currentMeterValue(fo.meterType.stealth)
+        predicted_stealth = total_stealth
+        if object_is_in_ion_storm(planet_id):
```

Comment:
The ion storm prediction is a rather complex issue-- we could know that an ion storm currently covers a planet but our last sighting of the planet (and corresponding meter readings) might or might not predate such.  Also, as a general matter we'll want to eventually do a really good job of predicting ion storm coverage (i.e. projecting trajectory from last observation), increasing the number of times when we might predict a planet to be currently covered by storm even if it was not at the last time we saw it.

So until such time as we got very confident of our ion storm predictions, I would probably suggest only using them to eliminate cases where we might otherwise warn about a prediction mismatch, not making a construction where a false prediction of ion storm coverage would trigger a warning (in that case, a debug statement at most, but I would even suggest against that for now, I think it would be too much noise).